### PR TITLE
[OvenPlayer] updating types, export types so developers can finnaly use them, Mark deleted method as unrequired, new types.

### PR DESCRIPTION
**Pull request by dygy**
### [OvenPlayer] updating types, export types so developers can finnaly use them, Mark deleted method as unrequired, new types.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<[url here](https://airensoft.gitbook.io/ovenplayer/api-reference/api)>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/ovenplayer/ovenplayer-tests.ts [hunk 1]
```diff
@@ -86,52 +88,72 @@ const providerName: string | null = player.getProviderName();
...
+player.once('play', data => {});
 
 // off(eventName: string): void;
-player.off("ready");
+player.off('ready');
 
 // remove(): void;
 player.remove();
+
+// @ts-expect-error: it's deprecated method, should throw error for newest users.
```
Comment (dygy) [inline types/ovenplayer/ovenplayer-tests.ts:147]: i'm sorry, i'm not sure if it's "deprecated", but i mean you can't find it in [github](https://github.com/search?q=repo%3AAirenSoft%2FOvenPlayer%20setCaption&type=code) or in docs. And i where unable to run it on my projects. Property is undefined.

Comment (SangwonOh) [reply types/ovenplayer/ovenplayer-tests.ts:147]: Perhaps the API that was attempted to be implemented during the initial design was not implemented well and is not working now.
And right now this API is not being used at all. I think the deprecated method is correct.

**Review by SangwonOh — APPROVED** [review]
@dygy Thanks! Very nice work! Helps a lot of peoples.

**Comment by SangwonOh** [comment]
Ready to merge

**Comment by dygy** [comment]
@SangwonOh i have small hotfix :D sorry https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66829/files

**Pull request merged**